File: smartmoneyserver/src/main/java/fpt/aptech/server/dto/icon/IconDto.java
package fpt.aptech.server.dto.icon;

/**
 * DTO (Data Transfer Object) cho thông tin của một icon.
 *
 * @param fileName Tên file đầy đủ (ví dụ: "icon_food.png"), dùng để lưu vào database.
 * @param url      URL đầy đủ (https) của icon, dùng để Flutter hiển thị.
 */
public record IconDto(
        String fileName,
        String url
) {
}

File: APP MOBILE/smart_money/lib/modules/category/models/icon_dto.dart
/// DTO chứa thông tin icon từ server.
/// Tương ứng: IconDTO.java (server) — GET /api/icons
/// fileName: tên file lưu trong database (VD: "icon_food.png")
/// url: URL đầy đủ Cloudinary để hiển thị (VD: "https://res.cloudinary.com/.../icon_food.png")
class IconDto {
  final String fileName; // tên file lưu vào DB khi tạo/sửa category
  final String url;      // URL đầy đủ để hiển thị ảnh

  const IconDto({
    required this.fileName,
    required this.url,
  });

  factory IconDto.fromJson(Map<String, dynamic> json) {
    return IconDto(
      fileName: json['fileName'] as String,
      url: json['url'] as String,
    );
  }
}

